src/main/java/edu/njit/jerse/ashe/services/MethodReplacementService.java

```diff
@@ -55,8 +56,8 @@ public static boolean replaceMethodInFile(String absoluteFilePath, String classN
 
         Path path = Paths.get(absoluteFilePath);
 
-        Optional<MethodSignature> methodSignatureOpt = JavaCodeParser.extractMethodSignature(newMethodCode);
-        if (methodSignatureOpt.isEmpty() || !isValidMethodSignature(methodSignatureOpt.get())) {
+        MethodSignature methodSignature = JavaCodeParser.extractMethodSignature(newMethodCode);
+        if (!isValidMethodSignature(methodSignature)) {
```

Comment:
Logically, I think it makes more sense for `isValidMethodSignature` to be a method of the `MethodSignature` class (probably named just `isValid`), rather than a static method of this class: being valid is a property of a `MethodSignature`.